hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/HazelcastClientCachingProvider.java

```diff
@@ -123,6 +127,27 @@ private HazelcastInstance getDefaultInstance() {
         return hazelcastInstance;
     }
 
+    /**
+     * Get an existing {@link HazelcastInstance} by {@code instanceName} or, if not found, create a new {@link HazelcastInstance}
+     * with default configuration and given {@code instanceName}.
+     *
+     * @param instanceName name by which to lookup existing {@link HazelcastInstance} or create new one.
+     * @return             a {@link HazelcastInstance} with the given {@code instanceName}
+     */
+    private HazelcastInstance getOrCreateByInstanceName(String instanceName) {
+        HazelcastInstance instance = HazelcastClient.getHazelcastClientByName(instanceName);
```

Comment:
Since there is a non atomic check then act; could it be that the the system ends up with multiple hz instances with the same name?